### PR TITLE
Re-order display fullscreen handling and ensure toggle_fullscreen works for maximized window (by Temmie)

### PR DESCRIPTION
**Pull request by oddbookworm**
Port of https://github.com/pygame/pygame/pull/3497 into pygame-ce

### CODE REVIEW

**Comment by robertpfeiffer** [comment]
No, @Temmie3754 was right and I was wrong. It works.

**Review by MyreMylar — APPROVED** [review]
Still LGTM after the switch over 👍

File: src_c/display.c [hunk 1]
```diff
@@ -1091,6 +1091,17 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
...
+                 * resize/bordered/hidden changes due to SDL ignoring those
+                 * changes if the window is fullscreen
+                 * See https://github.com/pygame/pygame/issues/2711 */
+                if (0 !=
+                    SDL_SetWindowFullscreen(
+                        win, sdl_flags & (SDL_WINDOW_FULLSCREEN |
+                                          SDL_WINDOW_FULLSCREEN_DESKTOP))) {
+                    return RAISE(pgExc_SDLError, SDL_GetError());
+                }
+
```
Comment (robertpfeiffer) [inline src_c/display.c:1104]: looks good

File: src_c/display.c [hunk 2]
```diff
@@ -2254,7 +2258,9 @@ pg_toggle_fullscreen(PyObject *self, PyObject *_null)
                              1) != 0) {
                 return NULL;
             }
-            win = SDL_CreateWindow(state->title, wx, wy, w, h, 0);
+            flags &= ~SDL_WINDOW_FULLSCREEN_DESKTOP;
+            /* SDL_WINDOW_FULLSCREEN_DESKTOP includes SDL_WINDOW_FULLSCREEN */
+            win = SDL_CreateWindow(state->title, wx, wy, w, h, flags);
```
Comment (robertpfeiffer) [inline src_c/display.c:2263]: :+1:

**Review by robertpfeiffer — APPROVED** [review]

**Review by MyreMylar — APPROVED** [review]
👍

**Pull request merged**